### ISSUE TITLE
HTTP normalization

### ISSUE THREAD

**Issue by krizhanovsky** [tempesta-tech/tempesta#2]
We need at least minimal HTTP requests normalization (like ngx_http_parse_complex_uri() from Nginx does it). The normalization must be implemented as a part of current HTTP FSM (to avoid double processing as in Nginx) and write normalized fields to appropriate part of TfwHttpReq.

Normalization depending on back-end server personality also must be done, however this is very customizable and expensive logic, so it should be possible to switch the functionality off. Thus it must be implemented in http_norm.h as plugable HTTP FSM states.

Basically, we shouldn't do the normalization if `Cache-Control: no-transform` is presented.

Depends on #902. Linked with #1207.

**Comment by krizhanovsky** [comment]
The issue was wrongly closed
